ludwig/trainers/trainer_lightgbm.py

```diff
@@ -981,23 +990,37 @@ def train_step(
         """
         from lightgbm_ray import RayLGBMClassifier, RayLGBMRegressor
 
-        output_feature = next(iter(self.model.output_features.values()))
+        output_feature = get_single_output_feature(self.model)
         gbm_sklearn_cls = RayLGBMRegressor if output_feature.type() == NUMBER else RayLGBMClassifier
 
+        # Prepare buffer for storing predictions on training data.
+        train_logits = torch.zeros(
+            (lgb_train.label.size, output_feature.num_classes)
+            if output_feature.type() == CATEGORY and output_feature.num_classes > 2
+            else (lgb_train.label.size,)
+        )
+
         gbm = gbm_sklearn_cls(n_estimators=boost_rounds_per_train_step, **params).fit(
             X=lgb_train,
             y=None,
             init_model=init_model,
             eval_set=[(s, n) for s, n in zip(eval_sets, eval_names)],
             eval_names=eval_names,
             # add early stopping callback to populate best_iteration
-            callbacks=[lgb.early_stopping(boost_rounds_per_train_step)],
+            callbacks=[lgb.early_stopping(boost_rounds_per_train_step), store_predictions(train_logits)],
             ray_params=self.ray_params,
             # NOTE: hummingbird does not support categorical features
```

Comment:
I would expect hummingbird to be able to support categorical features -- for GBMs, do we preprocess categorical values into ints (using the vocab)?

Reply:
We indeed convert category features to integers before passing them to LightGBM/hummingbird

Reply:
Would this comment be obsolete then (should we remove it?)?

Reply:
Removed